addon/-private/system/relationships/state/has-many.js

```diff
@@ -111,6 +111,32 @@ export default class ManyRelationship extends Relationship {
     super.removeCanonicalInternalModelFromOwn(internalModel, idx);
   }
 
+  unloadInverseInternalModelFromOwn(internalModel) {
+    super.unloadInverseInternalModelFromOwn(internalModel);
+
+    const canonicalIndex = this.canonicalState.indexOf(internalModel);
+
+    if (canonicalIndex !== -1) {
+      this.canonicalState.splice(canonicalIndex, 1);
+    }
+
+    const manyArray = this._manyArray;
+
+    if (manyArray) {
+      const currentState = manyArray.currentState;
+      const idx = currentState.indexOf(internalModel);
+
+      if (idx !== -1) {
+        currentState.splice(idx, 1);
+
+        manyArray.set('length', currentState.length);
+        manyArray.notifyPropertyChange('length');
+
+        this.internalModel.notifyHasManyRemoved(this.key, internalModel, idx);
```

Comment:
https://github.com/emberjs/ember.js/pull/15487 may fix, and if so the following would polyfill it.

```js
let maybeLengthDescriptor = Ember.ArrayProxy.proto().length;

if (typeof maybeLengthDescriptor === 'object' &&
  maybeLengthDescriptor !== null &&
  maybeLengthDescriptor.isDescriptor === true &&
  maybeLengthDescriptor._dependentKeys === undefined) {
  maybeLengthDescriptor._dependentKeys = ['arrangedContent.length'];
}
```